internal/pkg/local/save.go

```diff
@@ -2,105 +2,177 @@ package local
-	if err := m.manifest.PersistRecord(record); err != nil {
+	if err := w.manifest.PersistRecord(w.Record); err != nil {
 		return err
 	}
 
-	// Mkdir
-	dir := filepath.Join(m.ProjectDir(), record.RelativePath())
-	if err := os.MkdirAll(dir, 0755); err != nil {
-		errors.Append(fmt.Errorf("cannot create directory \"%s\": %w", dir, err))
-		return errors
+	// Save
+	w.createFiles()
+	w.transform()
+	if w.errors.Len() == 0 {
+		w.write()
+	}
+	return w.errors.ErrorOrNil()
+}
+
+func (w *modelWriter) createFiles() {
+	// meta.json
+	if metadata := utils.MapFromTaggedFields(model.MetaFileTag, w.Object); metadata != nil {
+		w.Metadata = filesystem.CreateJsonFile(w.Naming().MetaFilePath(w.Record.RelativePath()), metadata)
+	}
+
+	// config.json
+	if configuration := utils.MapFromOneTaggedField(model.ConfigFileTag, w.Object); configuration != nil {
+		w.Configuration = filesystem.CreateJsonFile(w.Naming().ConfigFilePath(w.Record.RelativePath()), configuration)
 	}
 
-	// Transform
-	configContent, err := m.transformOnSave(record, source)
-	if err != nil {
-		errors.Append(err)
+	// description.md
+	if description, found := utils.StringFromOneTaggedField(model.DescriptionFileTag, w.Object); found {
+		w.Description = filesystem.CreateFile(w.Naming().DescriptionFilePath(w.Record.RelativePath()), strings.TrimRight(description, " \r\n\t")+"\n")
 	}
+}
+func (w *modelWriter) allFiles() []*filesystem.File {
+	// Get all files
+	files := make([]*filesystem.File, 0)
 
-	// Write metadata file
-	if metadata := utils.MapFromTaggedFields(model.MetaFileTag, source); metadata != nil {
-		errPrefix := record.Kind().Name + " metadata"
-		if err := m.writeJsonFile(m.Naming().MetaFilePath(record.RelativePath()), metadata, errPrefix); err != nil {
-			errors.Append(err)
+	// meta.json
+	if jsonFile := w.Metadata; jsonFile != nil {
+		if file, err := jsonFile.ToFile(); err == nil {
+			files = append(files, file)
+		} else {
+			w.errors.Append(err)
 		}
 	}
 
-	// Write description file
-	if description, found := utils.StringFromOneTaggedField(model.DescriptionFileTag, source); found {
-		errPrefix := record.Kind().Name + " description"
-		if err := m.writeFile(m.Naming().DescriptionFilePath(record.RelativePath()), description, errPrefix); err != nil {
-			errors.Append(err)
+	// config.json
+	if jsonFile := w.Configuration; jsonFile != nil {
+		if file, err := jsonFile.ToFile(); err == nil {
+			files = append(files, file)
+		} else {
+			w.errors.Append(err)
 		}
 	}
 
-	// Write config file (can be modified by transformOnSave method)
-	if configContent == nil {
-		configContent = utils.MapFromOneTaggedField(model.ConfigFileTag, source)
+	// description.md
+	if file := w.Description; file != nil {
+		files = append(files, file)
+	}
+
+	// other
+	files = append(files, w.Extra...)
+	return files
+}
+
+func (w *modelWriter) transform() {
+	if err := w.mapper.BeforeSave(w.files); err != nil {
+		w.errors.Append(err)
+	}
+}
+
+func (w *modelWriter) write() {
```

Comment:
Hlavny rozdiel je, ze ak sa pocas zapisu nieco pokazi, tak sa to revertne.
Tato funkcia bola pre zapisovanie `blocks` v transformaciach, ... a robilo mi to problemy, ze je to rozdielne.
Teraz to plati pre vsetky modeli.

Najprv sa povodne subory zalohuju, prida sa im `.old`.
Zapisu sa nove subory.
Ak je vsetko OK, tak sa `.old` subory zmazu.
Ak nie, tak sa zmazu nove subory a obnovia sa `.old`.